api/src/main/java/org/hawkular/apm/api/model/trace/Node.java

```diff
@@ -324,16 +348,18 @@ public void addInteractionId(String id) {
      * @param scope The scope
      * @return The subset of correlation ids that are associated with the scope
      */
-    public List<CorrelationIdentifier> getCorrelationIds(Scope scope) {
+    public List<CorrelationIdentifier> findCorrelationIds(Scope... scope) {
         List<CorrelationIdentifier> ret = null;
 
         for (int i=0; i < correlationIds.size(); i++) {
```

Comment:
http://stackoverflow.com/a/256861

The for-each loop, introduced in release 1.5, gets rid of the clutter and the opportunity for error by hiding the iterator or index variable completely. The resulting idiom applies equally to collections and arrays:

```
   // The preferred idiom for iterating over collections and arrays 
     for (Element e : elements) {
          doSomething(e);
     }
```
When you see the colon (:), read it as “in.” Thus, the loop above reads as “for each element e in elements.” Note that there is no performance penalty for using the for-each loop, even for arrays. In fact, it may offer a slight performance advantage over an ordinary for loop in some circumstances, as it computes the limit of the array index only once. While you can do this by hand (Item 45), programmers don’t always do so.